src/browser-context-transport.ts

```diff
@@ -0,0 +1,147 @@
+import { Transport } from "./shared/transport.js";
+import { JSONRPCMessage, JSONRPCMessageSchema } from "./types.js";
+
+/**
+ * Transport implementation that uses the browser's MessageChannel API for communication
+ * between different browser contexts (iframes, workers, tabs, windows, etc.).
+ */
+export class BrowserContextTransport implements Transport {
+	private _port: MessagePort;
+	private _started = false;
+	private _closed = false;
+	
+	sessionId: string;
+
+	onmessage?: (message: JSONRPCMessage) => void;
+	onerror?: (error: Error) => void;
+	onclose?: () => void;
+
+	/**
+	 * Creates a new BrowserContextTransport using an existing MessagePort.
+	 * 
+	 * @param port The MessagePort to use for communication.
+	 * @param sessionId Optional session ID. If not provided, one will be generated.
+	 */
+	constructor(port: MessagePort, sessionId?: string) {
+		if (!port) {
+			throw new Error("MessagePort is required");
+		}
+		
+		this._port = port;
+		this.sessionId = sessionId || this.generateId();
+
+		// Set up event listeners
+		this._port.onmessage = (event) => {
+			try {
+				const message = JSONRPCMessageSchema.parse(event.data);
+				this.onmessage?.(message);
+			} catch (error) {
+				const parseError = new Error(`Failed to parse message: ${error}`);
+				this.onerror?.(parseError);
+			}
+		};
+
+		this._port.onmessageerror = (event) => {
+			const messageError = new Error(`MessagePort error: ${JSON.stringify(event)}`);
+			this.onerror?.(messageError);
+		};
+	}
+
+	/**
+	 * Internal method to generate a session ID.
+	 * This is separated so it can be used by static methods.
+	 */
+	private static generateSessionId(): string {
+		// Current timestamp as prefix (in base 36 for shorter string)
+		const timePrefix = Date.now().toString(36);
+		
+		// Random suffix
+		const randomSuffix = Math.random().toString(36).substring(2, 10);
+		
+		return `${timePrefix}-${randomSuffix}`;
+	}
```

Comment:
in the `sse` transport the crypto UUID is used. there is a [browser compatible implementation](https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID) that can be used

Reply:
You're right, updated